syntax = "proto3";

package playlist;

service Playlist {
  rpc Play (Empty) returns (Empty) {}
  rpc Pause (Empty) returns (Empty) {}
  rpc NextPrev (NextPrevRequest) returns (Empty) {}
  rpc AddSong (Song) returns (Empty) {}
  rpc GetAllSongs (Empty) returns (SongList) {}
  rpc UpdateConfig (Empty) returns (Empty) {}
  rpc UpdateSong (Song) returns (Empty) {}
  rpc DeleteSong (Song) returns (Empty) {}
}

message Song {
  string name = 1;
  int32 duration = 2;
}

message NextPrevRequest {
  enum Direction {
    PREV = 0;
    NEXT = 1;
  }

  Direction direction = 1;
}

message SongList {
  repeated Song songs = 1;
}

message Empty {}
